qa/tasks/cephfs/test_failover.py

```diff
@@ -340,6 +340,52 @@ def test_thrash(self):
 
         self.fs.wait_for_daemons(timeout=90)
 
+class TestFailoverBeaconHealth(CephFSTestCase):
+    CLIENTS_REQUIRED = 1
+    MDSS_REQUIRED = 1
+
+    def initiate_journal_replay(self, num_files=100):
+        """ Initiate journal replay by creating files and restarting mds server."""
+
+        self.config_set("mds", "mds_delay_journal_replay_for_testing", "5000")
+        self.mounts[0].test_files = [str(x) for x in range(num_files)]
+        self.mounts[0].create_files()
+        self.fs.fail()
+        self.fs.set_joinable()
+
+    def test_replay_beacon_estimated_time(self):
+        """
+        That beacon emits warning message with estimated time to complete replay
+        """
+        self.initiate_journal_replay()
+        self.wait_for_health("MDS_ESTIMATED_REPLAY_TIME", 60)
+        # remove the config so that replay finishes and the cluster
+        # is HEALTH_OK
+        self.config_rm("mds", "mds_delay_journal_replay_for_testing")
+        self.wait_for_health_clear(timeout=60)
+
+    def test_replay_estimated_time_accuracy(self):
+        self.initiate_journal_replay(250)
+        def replay_complete():
+            health = self.ceph_cluster.mon_manager.get_mon_health(debug=False, detail=False)
+            codes = [s for s in health['checks']]
+            return 'MDS_ESTIMATED_REPLAY_TIME' not in codes
+
+        def get_estimated_time():
+            while True:
+                health = self.ceph_cluster.mon_manager.get_mon_health(debug=False, detail=True)
+                codes = [s for s in health['checks']]
+                if 'MDS_ESTIMATED_REPLAY_TIME' in codes:
+                    message = health['checks']['MDS_ESTIMATED_REPLAY_TIME']['detail'][0]['message']
+                    time_duration = float(message.split(" ")[-1][0:-2])
+                    completion_percentage = float(message.split(" ")[2][0:-1])
+                    log.debug(f"MDS_ESTIMATED_REPLAY_TIME is present in health: {message}, duration: {time_duration}, completion_percentage: {completion_percentage}")
+                    if completion_percentage >= 50:
+                        return time_duration
+            time.sleep(1)
```

Comment:
Unless my eyes deceive me, this sleep is not in the `while` loop?

Reply:
Tha's correct - need to one level indented.